### DISCUSSION TITLE
Using MMAction2 for regression

### DISCUSSION THREAD

**Discussion by BobPirate6** [open-mmlab/mmaction2#2792] [Ideas]
Hi!
It might sound unusual, but would it be possible to use MMAction2 for video regression?
I've got a dataset of video ads and their performance scores. All prepared and normalized.

The idea is to take the part of MMAction2 that detects actions and train it to predict scores, whatever actions there are and whenever they happen, as we statistically know that some elements strongly correlate with ad performance, but we do not need to classify them, only to train a model to detect them and connect to a singl number - a score.

Where do I start? How to edit the pipeline to switch from classification to regression?
I've done research using CNN+RNN networks for this purpose, but they all lack the ability to determine and track complex spacio-temporal actions, that sometimes can not be classified as "jump" or "judo throw".

**Comment by rlleshi** [comment]
I guess there are two main modifications to be made here:

- Replace the final classification layer with a regression one (i.e. a fully-connected layer with a single output neuron)
- Replace the classification loss, with a regression loss, e.g. mean squared error

checkout [the docs](https://mmaction2.readthedocs.io/en/latest/) to see how to add a new loss for example here
